app/src/main/java/com/rockspin/androiddevtest/APICommThread.java

```diff
@@ -0,0 +1,33 @@
+package com.rockspin.androiddevtest;
+
+import android.content.Intent;
+
+import java.util.List;
+
+import retrofit2.Response;
+
+/**
+ * Created by SG on 08.01.2017 г..
+ */
+public class APICommThread extends Thread
+{
+    @Override
```

Comment:
My point was that you have threading in java itself. Why should you rely on 3rd party libs. You also have several ways for communication between threads natively in java. Again why should you relay on 3rd party libs? That's why I used java Thread class.